### DISCUSSION TITLE
Model-based control

### DISCUSSION THREAD

**Discussion by NarimanNiknejad** [bitcraze/discussions#1450] [Q&A] [answered]
Hi,

I'm implementing a model-based controller for Crazyflie and have encountered an issue. My model's u(t) is thrust and angular rates, but the Crazyflie's motors require velocity inputs. Is there a conversion step I'm missing to bridge this gap? Any insights on translating thrust and angular rates to motor velocities would be greatly appreciated. Thanks in advance for your help!

**Comment by ataffanel** [comment, marked as answer]
This is a step that requires system identification on the real platform. This has been done by many groups over time. An [issue](https://github.com/bitcraze/crazyflie-simulation/issues/60) in the simulation repos is talking about it. IMRCLab has published a [repos with system identification](https://github.com/IMRCLab/crazyflie-system-id) of the Crazyflie, it does contain PWM-to-thrust as well as RPM-to-thrust.

**Reply by NarimanNiknejad** [reply]
Thank you so much! It is very helpful.
